server/zio-http-server/src/main/scala/sttp/tapir/server/ziohttp/ZioWebSockets.scala

```diff
@@ -0,0 +1,65 @@
+package sttp.tapir.server.ziohttp
+
+import io.netty.buffer.Unpooled
+import sttp.capabilities.zio.ZioStreams
+import sttp.capabilities.zio.ZioStreams.Pipe
+import sttp.tapir.{DecodeResult, WebSocketBodyOutput, WebSocketFrameDecodeFailure}
+import sttp.ws.WebSocketFrame
+import zhttp.socket.{Socket, WebSocketFrame => ZioWebSocketFrame}
+import zio.stream.Stream
+
+private[ziohttp] object ZioWebSockets {
+  def pipeToBody[REQ, RESP](
+      pipe: Pipe[REQ, RESP],
+      o: WebSocketBodyOutput[Pipe[REQ, RESP], REQ, RESP, _, ZioStreams]
+  ): Socket[Any, Throwable, ZioWebSocketFrame, ZioWebSocketFrame] = {
+    Socket.fromFunction[ZioWebSocketFrame] { (zFrame: ZioWebSocketFrame) =>
+      val sttpFrames = Stream.succeed(zFrameToFrame(zFrame))
+      val ignorePongs = optionallyIgnorePong(sttpFrames, o.ignorePong)
```

Comment:
is concatenation, auto-pings & auto-pongs left for later or are there some problems to implement this?

Reply:
I created a feature request in zio-http - https://github.com/dream11/zio-http/issues/892 since it's seems not possible now